File: py_pkg/py_pkg/param_node.py
import rclpy
import rclpy.node

class SimpleParamNode(rclpy.node.Node):
    def __init__(self):
        super().__init__('param_node')
        time_period = 1 
        self.timer_ = self.create_timer(time_period, self.timer_callback)
        self.declare_parameter('my_param', 'world')
    def timer_callback(self):
        my_param = self.get_parameter('my_param').get_parameter_value().string_value
        self.get_logger().info("Hello " + my_param)        

        #  my_new_param = rclpy.Parameter('my_param', rclpy.Parameter.Type.STRING, 'world')
        #  all_new_params = [my_new_param]
        #  self.set_parameters(all_new_params)

def main():
    rclpy.init()
    node = SimpleParamNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ ==  '__main__':
    main()

File: cpp_pkg/src/param_node.cpp
#include <chrono>
#include <functional>
#include <memory>
#include <rclcpp/executors.hpp>
#include <rclcpp/node.hpp>
#include <rclcpp/rclcpp.hpp>
#include <rclcpp/timer.hpp>
#include <rclcpp/utilities.hpp>
#include <string>

using namespace std::chrono_literals;

class SimpleParamClass : public rclcpp::Node {

public:
  SimpleParamClass() : Node("param_node") {
    this->declare_parameter<std::string>("my_param", "world");
    timer_ = this->create_wall_timer(
        1000ms, std::bind(&SimpleParamClass::respond, this));
  }
  void respond() {
    this->get_parameter("my_param", param_string_);
    RCLCPP_INFO(this->get_logger(), "Hello %s", param_string_.c_str());
  }

private:
  std::string param_string_;
  rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char *argv[]) {
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<SimpleParamClass>());
  rclcpp::shutdown();
  return 0;
}
